src/ORM/Rule/ExistsIn.php

```diff
@@ -98,6 +109,11 @@ public function __invoke(EntityInterface $entity, array $options)
             return true;
         }
 
+        if ($this->_options['allowSqlNulls'] === true
+            && $this->_checkPartialSchemaNulls($entity, $source) === true
+        ) {
+            return true;
```

Comment:
I cannot see what you are trying to do in this PR. This code would return `true` against key pairs such as the following:
```php
[
    'parent_id' => null,
    'site_id' => 99999, -- Non existent
]
```

But seems to be intended. Because `_fieldsAreNull()` is enough to return `true` against key pairs such as the following:
```php
[
    'parent_id' => null,
    'site_id' => 1, -- Existent
]
```

So how does this feature benefit you? I think you might want to explain it in more detail.